#! /bin/sh /usr/share/dpatch/dpatch-run
## 40_replace_egrep_fgrep_sigs.dpatch by David Weinehall <tao@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Replace egrep by grep -E, fgrep by grep -F, and numerical signals
## DP: by signal names

@DPATCH@
diff -ur texinfo-4.6-old/makeinfo/tests/cond texinfo-4.6/makeinfo/tests/cond
--- texinfo-4.6-old/makeinfo/tests/cond	2000-12-22 17:02:32.000000000 +0200
+++ texinfo-4.6/makeinfo/tests/cond	2004-06-25 21:49:28.000000000 +0300
@@ -5,28 +5,28 @@
 
 # Default Info output.
 ../makeinfo -o cond.out $srcdir/cond.txi || exit 1
-egrep 'This is (ifnothtml|ifinfo|ifnottex) text' cond.out >/dev/null \
+grep -E 'This is (ifnothtml|ifinfo|ifnottex) text' cond.out >/dev/null \
   || exit 2
-test `fgrep ' text.' cond.out | wc -l` -eq 3 || exit 3
+test `grep -F ' text.' cond.out | wc -l` -eq 3 || exit 3
 
 # Default HTML output.
 ../makeinfo --no-split --html -o cond.out $srcdir/cond.txi || exit 1
-egrep 'This is (html|ifhtml|ifnotinfo|ifnottex) text' cond.out >/dev/null \
+grep -E 'This is (html|ifhtml|ifnotinfo|ifnottex) text' cond.out >/dev/null \
  || exit 2
-test `fgrep ' text.' cond.out | wc -l` -eq 4 || exit 3
+test `grep -F ' text.' cond.out | wc -l` -eq 4 || exit 3
 
 # --ifhtml off, --ifinfo off, --iftex off.
 ../makeinfo --no-ifhtml --no-ifinfo --no-iftex -o cond.out $srcdir/cond.txi || exit 1
-egrep 'This is ifnot(html|info|tex) text' cond.out >/dev/null \
+grep -E 'This is ifnot(html|info|tex) text' cond.out >/dev/null \
  || exit 2
-test `fgrep ' text.' cond.out | wc -l` -eq 3 || exit 3
+test `grep -F ' text.' cond.out | wc -l` -eq 3 || exit 3
 
 # Do we really need to test all the other permutations?
 
 # --ifhtml on, --ifinfo on, --iftex on.
 ../makeinfo --ifhtml --ifinfo --iftex -o cond.out $srcdir/cond.txi || exit 1
-egrep 'This is (html|ifhtml|ifinfo|tex|iftex) text' cond.out >/dev/null \
+grep -E 'This is (html|ifhtml|ifinfo|tex|iftex) text' cond.out >/dev/null \
  || exit 2
-test `fgrep ' text.' cond.out | wc -l` -eq 5 || exit 3
+test `grep -F ' text.' cond.out | wc -l` -eq 5 || exit 3
 
 rm -f cond.out cond.info
diff -ur texinfo-4.6-old/util/fixfonts texinfo-4.6/util/fixfonts
--- texinfo-4.6-old/util/fixfonts	1993-04-16 13:35:07.000000000 +0300
+++ texinfo-4.6/util/fixfonts	2004-06-25 21:48:37.000000000 +0300
@@ -33,7 +33,7 @@
 tempfile2="${TMPDIR}/circ2$$"
 
 # EXIT SIGHUP SIGINT SIGQUIT SIGTERM
-#trap 'rm -f "${tempfile}" "${tempfile2}"' 0 1 2 3 15
+#trap 'rm -f "${tempfile}" "${tempfile2}"' 0 HUP INT QUIT TERM
 
 # Find all the fonts with names that include `circle'.
 (cd "${texfontdir}"; find . -name '*circle*' -print > "${tempfile}")
diff -ur texinfo-4.6-old/util/gen-dir-node texinfo-4.6/util/gen-dir-node
--- texinfo-4.6-old/util/gen-dir-node	2003-02-01 16:18:59.000000000 +0200
+++ texinfo-4.6/util/gen-dir-node	2004-06-25 21:47:56.000000000 +0300
@@ -79,7 +79,7 @@
 ### then generate entries for those in the same way, putting the info for 
 ### those at the end....
 
-infofiles=`(cd ${INFODIR}; /bin/ls | grep -v '\-[0-9]*$' | egrep -v '^dir$|^dir\.info$|^dir\.orig$')`
+infofiles=`(cd ${INFODIR}; /bin/ls | grep -v '\-[0-9]*$' | grep -E -v '^dir$|^dir\.info$|^dir\.orig$')`
 
 # echoing gets clobbered by backquotes; we do it the hard way...
 lines=`wc $SKELETON | awk '{print $1}'`
@@ -166,8 +166,8 @@
 
 # Sort remaining files by INFO-DIR-SECTION.
 prevsect=
-filesectdata=`(cd ${INFODIR}; fgrep INFO-DIR-SECTION /dev/null ${infofiles} | \
-	      fgrep -v 'INFO-DIR-SECTION Miscellaneous' | \
+filesectdata=`(cd ${INFODIR}; grep -F INFO-DIR-SECTION /dev/null ${infofiles} | \
+	      grep -F -v 'INFO-DIR-SECTION Miscellaneous' | \
 	      sort -t: -k2 -k1 | tr ' ' '_')`
 for sectdata in ${filesectdata}; do
   file=`echo ${sectdata} | cut -d: -f1`
diff -ur texinfo-4.6-old/util/texi2dvi texinfo-4.6/util/texi2dvi
--- texinfo-4.6-old/util/texi2dvi	2004-06-24 15:16:18.000000000 +0300
+++ texinfo-4.6/util/texi2dvi	2004-06-25 21:47:07.000000000 +0300
@@ -220,7 +220,7 @@
 
 # Prepare the temporary directory.  Remove it at exit, unless debugging.
 if test -z "$debug"; then
-  trap "cd / && rm -rf $tmpdir" 0 1 2 15
+  trap "cd / && rm -rf $tmpdir" 0 HUP INT TERM
 fi
 
 # Create the temporary directory with strict rights


